app/server/lib/PermissionInfo.ts

```diff
@@ -219,7 +219,7 @@ function evaluateRule(ruleSet: RuleSet, input: AclMatchInput): PartialPermission
         // Anything it would explicitly allow, no longer allow through this rule.
         // Anything it would explicitly deny, go ahead and deny.
         pset = mergePartialPermissions(pset, mapValues(rule.permissions, val => (val === 'allow' ? "" : val)));
-        log.warn("ACLRule for %s failed: %s", ruleSet.tableId, e.message);
+        log.warn("ACLRule for %s (`%s`) failed: %s", ruleSet.tableId, rule.aclFormula, e.stack);
```

Comment:
This warning was really half-baked, thanks for giving it some love. One important piece of context missing is what document the rule is in. There's more, but that's a big one. I think it might be fairly straightforward to add some optional context in `ACLMatchInput` and then have `GranularAccess.inputs` set that up, but I could be wrong. Leaving it to you whether you want to look into it, this is already an improvement.